### PR TITLE
Report unnecessary_parenthesis around Literal and PrefixedIdentifier

### PR DESCRIPTION
**Pull request by oprypin**
Fixes a few false ~~positives~~ negatives that I noticed when mass-editing code and gratuitously sprinkling parentheses everywhere

### CODE REVIEW

**Review by bwilkerson — APPROVED** [review]

File: test_data/rules/unnecessary_parenthesis.dart [hunk 1]
```diff
@@ -145,8 +145,25 @@ main() async {
   a2 = (1 == 1); // OK
   a2 = (1 == 1) || "".isEmpty; // OK
   var a3 = (1 + 1); // LINT
+
+  // Tests for Literal and PrefixedIdentifier
+  var a4 = (''); // LINT
```
Comment (pq) [inline test_data/rules/unnecessary_parenthesis.dart:150]: You mention only false positives but these tests validate (new) reports. Is that right? Is this actually fixing false negatives?

Comment (oprypin) [reply test_data/rules/unnecessary_parenthesis.dart:150]: Oh sorry, I mixed up the two. I am only fixing false negatives here. I.e. only adding more complaint cases.

**Comment by oprypin** [comment]
Note: this will cause a few new lint violations in google3, but I'll handle those

**Review by pq — APPROVED** [review]

**Comment by pq** [comment]
Good deal. Thanks!

**Pull request merged**